engine/uState.ml

```diff
@@ -226,7 +226,11 @@ let process_universe_constraints ctx cstrs =
                     let levels = Univ.Universe.levels l in
                     let fold l' local =
                       let l = Univ.Universe.make l' in
-                      if Univ.Level.is_small l' || is_local l' then
+                      if Univ.Level.is_prop r' then begin
+                        vars := Univ.LMap.add l' None !vars;
```

Comment:
This seems wrong. instantiate_variable should be sufficient.

Reply:
I did not change the `rigid` status given to sort variables in `Evarutil.new_type_evar` (and ...), so, the level `l'` is not added to `vars`/`uctx_univ_variables` (in `new_univ_variable`) and `instantiate_variable` would fail without the explicit extra `Univ.LMap.add`.

More fundamentally, I did not understand whether the status `flexible` was reserved to universe polymorphism or if it could have been used in the current (non-polymoprhic) situation.

Reply:
Making arbitrary levels flexible is definitely wrong.
rigid is for `Type` (without annotation) and named levels, I think everything else should be flexible.
Evarutil.new_type_evar is passed the desired rigidity so I don't understand what you mean there.

Reply:
define_evar_as_sort uses univ_rigid, this seems questionable.

Reply:
I updated the PR, fixing `define_evar_as_sort` and now interpreting flexible as existential (the usual way then).